plugin/phpactor.vim

```diff
@@ -145,24 +145,50 @@ endfunction
 """"""""""""""""""""""""
 " Extract method
 """"""""""""""""""""""""
-function! phpactor#ExtractMethod()
-    let selectionStart = phpactor#_selectionStart()
-    let selectionEnd = phpactor#_selectionEnd()
 
-    call phpactor#rpc("extract_method", { "path": phpactor#_path(), "offset_start": selectionStart, "offset_end": selectionEnd, "source": phpactor#_source()})
+function! phpactor#ExtractMethod(...)
+    let positions = {}
+
+    if 0 == a:0 " Visual mode - backward compatibility
+        let positions.start = phpactor#_selectionStart()
+        let positions.end = phpactor#_selectionEnd()
+    elseif a:1 ==? 'v' " Visual mode
+        let positions.start = phpactor#_selectionStart()
+        let positions.end = phpactor#_selectionEnd()
+    else " Linewise or characterwise motion
+        let linewise = 'line' == a:1
+
+        let positions.start = s:getStartOffsetFromMark("'[", linewise)
+        let positions.end = s:getEndOffsetFromMark("']", linewise)
```

Comment:
If this logic is the same as below, can we extract a common method?

Reply:
I guess the difference is that expression can work with just a single offset, not sure if method works like that - in theory it could extract the expression under the cursor.

Reply:
Yes the important is not the two lines but the logic block just above.
And we don't handle expression and method extraction the same way.
Also rule of three: if there is only those 2 occurences I prefer to wait until refactoring.